handout/modules/introduction.md

```diff
@@ -0,0 +1,89 @@
+
+The property `imports` expects an array of modules. Here's where we define the pieces of our puzzle (our application). The property `declarations` expects an array of components and pipes that are part of the module. The `bootstrap` property is where we define the root component of our module. Even though this property is also an array, 99% of the time we are going to define only one component.
+
+> There are very special circumstances where more than one component may be required to bootstrap a module but we are not going to cover those edge cases here.
+
+Here's how a basic module made up of just one component would look like:
+
+_app/app.component.ts_
+
+```js
+import { Component } from "@angular/core";
+
+@Component({
+  selector: 'rio-app',
+  template: '<h1>My Angular 2 App</h1>'
+})
+export class AppComponent {}
+```
+
+_app/app.module.ts_
+
+```js
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
+import { AppComponent } from './app.component';
+
+@NgModule({
+  imports: [BrowserModule],
+  declarations: [AppComponent],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+The file _app.component.ts_ is just a "hello world" component, nothing interesting there. In the other hand, the file _app.module.ts_ is following the structure that we've seen before for defining a module but in this case, we are defining the modules and components that we are going to be using.
+
+The first thing that we notice is that our module is importing the `BrowserModule` as an explicit dependency. The `BrowserModule` is a built-in module that exports basic directives, pipes and services like `NgFor` and `NgIf`. Unlike previous versions of Angular 2, we have to explicitly import those dependencies to be able to use directives like `*ngFor` or `*ngIf` in our templates.
+
+Given that the root (and only) component of our module is the `AppComponent` we have to list it in the `bootstrap` array. Because in the `declarations` property we are supposed to defined **all** the components or pipes that make up our application, we have to define again the `AppComponent` there. 
+
+Before moving on, there's an important clarification to make. **There are two types of modules, root modules and feature modules**.
+
+In the same way that in a module we have one root component and possible many secondary components, **in an application we only have one root module and zero or many feature modules**. To be able to bootstrap our application, Angular needs to know which one is the root module. An easy way to identify a root module is by looking at the `imports` property of its `NgModule` decorator. If the module is importing the `BrowserModule` then it's a root module, if instead is importing the `CommonModule` then is a feature module.
```

Comment:
"...and many possible secondary components..."
"...importing the `CommonModule` then _it_ is a feature module."